DigitalLearningSolutions.Web/Controllers/LearningMenuController/LearningMenuController.cs

```diff
@@ -153,6 +155,58 @@ public IActionResult Diagnostic(int customisationId, int sectionId)
             return View("Diagnostic/Diagnostic", model);
         }
 
+        [Route("/LearningMenu/{customisationId:int}/{sectionId:int}/Diagnostic/Content")]
+        public IActionResult DiagnosticContent(int customisationId, int sectionId, List<int> checkedTutorials)
+        {
+            var candidateId = User.GetCandidateId();
+            var centreId = User.GetCentreId();
+            var diagnosticContent = diagnosticAssessmentService.GetDiagnosticContent(customisationId, sectionId);
+
+            if (diagnosticContent == null || centreId == null)
+            {
+                logger.LogError(
+                    "Redirecting to 404 as section/centre id was not found. " +
+                    $"Candidate id: {candidateId}, customisation id: {customisationId}, " +
+                    $"centre id: {centreId?.ToString() ?? "null"}, section id: {sectionId}");
+                return RedirectToAction("StatusCode", "LearningSolutions", new { code = 404 });
+            }
+
+            if (checkedTutorials.Except(diagnosticContent.Tutorials).Any())

```

Comment:
I think this belongs in a service, not in the controller. There's also another check we need; if diagObjSelect is not true then the selected tutorials should be ignored and all tutorials used. I don't think you have any handling of that at the moment? I think splitting the `DiagnosticAssessmentService` into two services; `DiagnosticAssessmentDataService` which handles just fetching the data like the current service does, and `DiagnosticAssessmentService` which handles any processing of the data like this. What do you think? 🙂

Reply:
I'm not sure what we want to do in the new `DiagnosticAssessmentService`, would it have a method taking the `checkedTutorials` and `diagnosticContent` then return a `bool`? How about using a helper class?

Reply:
The new service should be a wrapper around the old one. So it should have a method to get the diagnostic assessment and one to get the diagnostic assessment content, like the current service does. The first method will just call the same method on the data service. But the second will take the checked tutorials and do the checks described after fetching the assessment content. Does that make sense? 🙂